app/src/main/java/ch/sdp/vibester/fragment/SearchUserFragment.kt

```diff
@@ -113,7 +115,7 @@ class SearchUserFragment : Fragment(R.layout.fragment_search_user), OnItemClickL
     }
 
     override fun onItemClick(position: Int) {
-        val intent = Intent(requireActivity(), PublicProfileActivity::class.java)
+        val intent = Intent(activity, PublicProfileActivity::class.java)
         intent.putExtra("UserId", users[position].uid)
         intent.putExtra("ScoresOrFollowing", R.string.profile_following.toString())
         startActivity(intent)
```

Comment:
Same thing as what Maxime had said, could you look into refactoring this with the Intent Switcher?